apollo-router/src/plugins/telemetry/filtering_layer.rs

```diff
@@ -0,0 +1,141 @@
+use std::collections::HashMap;
+use std::collections::HashSet;
+
+use tracing::field;
+use tracing::Level;
+use tracing::Subscriber;
+use tracing_core::Field;
+use tracing_subscriber::Layer;
+
+// Specific attributes for logging
+pub(crate) const SPECIFIC_ATTRIBUTES: [&str; 4] = [
+    "request",
+    "response_headers",
+    "response_body",
+    "operation_name",
+];
+
+const SUBGRAPH_ATTRIBUTE_NAME: &str = "subgraph";
+
+#[derive(Debug, Clone, Default, PartialEq)]
+pub(crate) struct AttributesToExclude {
+    pub(crate) supergraph: HashSet<String>,
+    pub(crate) all_subgraphs: HashSet<String>,
+    pub(crate) subgraphs: HashMap<String, HashSet<String>>,
+}
+
+pub(crate) struct FilterLayer {
+    attributes_to_exclude: Option<AttributesToExclude>,
+}
+
+impl FilterLayer {
+    pub(crate) fn new(attributes_to_exclude: Option<AttributesToExclude>) -> Self {
+        Self {
+            attributes_to_exclude,
+        }
+    }
+}
+
+impl<S> Layer<S> for FilterLayer
+where
+    S: Subscriber + Send + Sync + 'static,
+{
+    fn enabled(
+        &self,
+        metadata: &tracing::Metadata<'_>,
+        ctx: tracing_subscriber::layer::Context<'_, S>,
+    ) -> bool {
+        // The logic is splitted between `enabled` and `event_enabled` because we can't evaluate the value of an attribute in this method
+        // This method is evaluated only once for a specific callsite, which means if I return false here I won't re-enter in this method everytime for this event
+        // Which is better for performance. So for supergraph level we can use this method, for subgraph level we have to know the value of subgraph attribute
+        if metadata.level() == &Level::DEBUG && metadata.is_event() {
```

Comment:
why is attribute filtering only active at the `DEBUG` level?

Reply:
Because these logs are always in DEBUG, but following a discussion with Bryn it will be in INFO

Reply:
Agree that seems strange. Should be mentioned in the documentation, but I don't really understand why it exists. Also, if DEBUG is special for some reason, don't we also want to process when level is TRACE?

Reply:
I deleted the DEBUG level condition in the process